### ISSUE TITLE
Filling missing with empty array removes rows

### ISSUE THREAD

**Issue by david-cortes** [pola-rs/polars#5146]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

If I have a polars series with missing values and I try to replace those missing values with empty lists/arrays, the result from the method `fill_nan` will return a series with fewer rows than the original one.

### Reproducible example

```python
import numpy as np, polars as pl
s = pl.Series(np.array([None, None], dtype="O"))
s
```
```
shape: (2,)
Series: '' [o][object]
[
	None
	None
]
```
```python
s.fill_nan(fill_value=pl.lit(np.array([], dtype="O")))
```
```
shape: (0,)
Series: '' [o][object]
[
]
```

### Expected behavior

The output from `fill_nan` should always have the same number of rows as the input.

### Installed versions

<details>

```
---Version info---
Polars: 0.14.18
Index type: UInt32
Platform: Linux-5.19.0-2-amd64-x86_64-with-glibc2.35
Python: 3.10.4 (main, Mar 31 2022, 08:41:55) [GCC 7.5.0]
---Optional dependencies---
pyarrow: 9.0.0
pandas: 1.5.0
numpy: 1.23.3
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: <not installed>
```

</details>

**Comment by alexander-beedie** [comment]
I think there is probably a real bug here (I can't think of a scenario where it makes sense that you'd get zero values back ;), but it's also a very odd way to initialize a frame - is this connected to a clearer/concrete use-case? 

For instance, initializing as a numpy object array means that the `None` values you see in the series are _not_ actually frame-native `null` values, and consequently wouldn't be affected by the subsequent call to `fill_null`.

**Comment by david-cortes** [comment]
It was not an intended initialization - I was trying to fill missing values in a `list[dtype]` column with empty lists but could not find any way to do it with polars.

**Comment by ritchie46** [comment]
> `fill_nan` is designed to work on float dtypes. I think, if anything, OP's code should raise an error that `fill_nan()` is not implement for series of dtype `pl.Object`.
> 
> I notice that `fill_nan` works on some dtypes and not others:
> 
> ```python
> pl.Series([1, 2, 3]).fill_nan(0)                                                     # returns original series
> pl.Series(['1', '2', '3']).fill_nan(0)                                               # returns original series
> pl.Series([1, 2, 3], dtype=pl.Object).fill_nan(0)                                    # raises exception
> pl.Series([[1], [2], [3]]).fill_nan([0])                                             # returns original series
> pl.Series([date(2022,1,1), date(2022,1,2), date(2022,1,3)]).fill_nan(date(2022,1,4)) # returns original series
> pl.Series([date(2022,1,1), date(2022,1,2), date(2022,1,3)]).fill_nan(0)              # returns `i64` array of days since 1970
> ```

Good observation. I agree.

**Comment by ritchie46** [comment]
Fix in #5185. @david-cortes are you using numpy arrays as objects, or was this simply to show the bug?
As doing so is probably not the best way to do things, maybe we can improve what you are doing.

**Comment by david-cortes** [comment]
@ritchie46 Thanks for fixing it. This was just an example - my original use-case was about filling missing values in `list[<type>]` columns with empty lists of the same `<type>`.
